servicenow/stackstate_checks/servicenow/servicenow.py

```diff
@@ -137,23 +142,26 @@ def _process_components(self, instance_config, timeout):
 
         state = self._collect_components(instance_config, timeout)
 
-        for component in state['result']:
-            component = self.filter_empty_metadata(component)
-            identifiers = []
-            comp_name = component['name']
-            comp_type = component['sys_class_name']
-            external_id = component['sys_id']
-
-            if 'fqdn' in component and component['fqdn']:
-                identifiers.append("urn:host:/{}".format(component['fqdn']))
-            if 'host_name' in component and component['host_name']:
-                identifiers.append("urn:host:/{}".format(component['host_name']))
-            identifiers.append("urn:host:/{}".format(comp_name))
-            identifiers.append(external_id)
-            data = {"identifiers": identifiers, "tags": instance_tags}
-            data.update(component)
-
-            self.component(external_id, comp_type, data)
+        if "result" in state:
```

Comment:
I think you also have to check the contents of `result`. In the error log it looked like it ended up being a string, is that still possible now?

Reply:
I am already checking the content of result is `dictionary` or not.